Lib/test/test_io.py

```diff
@@ -4414,15 +4409,11 @@ def test_pickling(self):
                         with self.assertRaisesRegex(TypeError, msg):
                             pickle.dumps(f, protocol)
 
-    @unittest.skipIf(
-        support.is_emscripten, "fstat() of a pipe fd is not supported"
-    )
+    @unittest.skipIf(support.is_emscripten, "Corrupts memory")
```

Comment:
This is a little concerning as a warning :-) I presume this is an emscripten issue with no obvious resolution in sight? If so, making it clear from the skip message that is the cause would be preferable:
```suggestion
    @unittest.skipIf(
        support.is_emscripten, 
        "Emscripten corrupts memory when calling fstat() on a pipe fd"
    )
```

Reply:
It's an emscripten bug. I haven't debugged it yet so I couldn't tell you how hard it is to fix. I'll try to look into it soon it's the next thing on my list.

Reply:
It's an emscripten bug. I haven't debugged it yet so I couldn't tell you how hard it is to fix. I'll try to look into it soon it's the next thing on my list.

Reply:
actual problem here is more like "Emscripten corrupts memory when writing to nonblocking fd" I think. I fixed calling fstat on pipe fd in https://github.com/emscripten-core/emscripten/pull/23306/